catalog/async/catalog-async-data/src/main/java/org/codice/ddf/catalog/async/data/impl/LazyProcessResourceImpl.java

```diff
@@ -84,9 +92,21 @@ public LazyProcessResourceImpl(String metacardId, Supplier<Resource> resourceSup
    * <p>This will load the resource.
    */
   @Override
-  public InputStream getInputStream() throws IOException {
+  public synchronized InputStream getInputStream() throws IOException {
     loadResource();
-    return inputStream;
+    if (inputStreamDataCache == null) {
+      if (inputStream == null) {
+        throw new IOException(String.format("Tried to get input stream for %s but was null", name));
+      }
+      inputStreamDataCache =
+          new TemporaryFileBackedOutputStream(FILE_BACKED_OUTPUT_STREAM_THRESHOLD);
+      IOUtils.copyLarge(inputStream, inputStreamDataCache);
```

Comment:
❗️ We should close `inputStream` right away as it would have been fully read after the copy. I am wondering if we should be storing `inputStream` as an attribute now it is only used to create the temporary file backed output stream. Maybe the logic to create that output stream should be done inside the `populateProcessResource()` method.

Reply:
I'll add the line back to close the input stream right after the copy. I discussed when to create the ouputstream with @lessarderic and we thought, since copying the data from the input stream to a temporary file takes quite a bit of time, that we should only write to that file if we know that the object handler is going to need a copy of the input stream.

Reply:
That is also what I was thinking but I forgot about the fact that the `loadResource()` was called from other places too.